GECO/Login.aspx.vb

```diff
@@ -32,52 +41,100 @@ Partial Class Login
 
     Protected Sub btnSignIn_Click(sender As Object, e As EventArgs) Handles btnSignIn.Click
         SignIn()
+
+    End Sub
+
+    Protected Sub Delay_SignIn()
+        ' btnSignIn.Enabled = False
+        isThrottled = True
+        Debug.Print("Delay " & failedSignInAttempts)
+        Thread.Sleep(20000)
+        isThrottled = False
+        ' btnSignIn.Enabled = True
     End Sub
 
     Protected Sub SignIn()
         lblMessage.Visible = False
         lblUnconfirmed.Visible = False
 
-        Dim gecoUser As New GecoUser
-        Dim userSession As New UserSession
+        'Static timeoutTimer As New CountDownTimer
+        ' set timer for 1 minutes
+        'timeoutTimer.SetTime(1, 0)
 
-        Dim loginResult As LoginResult = LogInUser(txtUserId.Text, txtPassword.Text, chkRememberMe.Checked, gecoUser, userSession)
+        ' Set the action each time the timer ticks down
+        timeoutTimer.TimeChanged = New Action(Sub()
```

Comment:
I'm a little bit confused here on why the variable `lblMessage.Text` does not change. I know that the timeoutTimer.TimeChanged is working properly since it outputs the timer count down correctly. 
I am creating a Sub procedure and extend it into an Action(), and from there the CountDownTimer object would invoke it (`TimeChanged?.Invoke()`)

Reply:
Changed so that this does not matter anymore